sys/net/link_layer/Kconfig.lorawan

```diff
@@ -0,0 +1,518 @@
+        the network.
+
+config LORAMAC_DEFAULT_ADR
+    bool "Enable ADR" if LORAMAC_ADVANCED_OPTIONS
+    depends on USEPKG_SEMTECH_LORAMAC || USEMODULE_RN2XX3
+    help
+        Enable or disable Adaptive Data Rate (ADR). If enabled the end node will
+        inform the network server about the status of ADR using the ADR field in
+        uplink data packet. The network server will then optimize the data rate
+        and the transmission power of the end node based on the information
+        collected from the network.
+
+config LORAMAC_DEFAULT_LINKCHK
+    int "Link check interval (in seconds)" if LORAMAC_ADVANCED_OPTIONS
+    depends on USEMODULE_RN2XX3
+    default 0
+    help
+        Configure the default link check interval (in seconds). 0 means the link
+        check process is disabled. The link check process is used by an
+        end-device to validate its connectivity with the network. The frame has
+        no payload. When a `LinkCheckReq` is received by the network server via
+        one or multiple gateways, it responds with a `LinkCheckAns` MAC command.
+
+config LORAMAC_DEFAULT_RX1_DELAY
+    int "RX1 window delay (in milliseconds)" if LORAMAC_ADVANCED_OPTIONS
+    depends on USEMODULE_GNRC_LORAWAN || USEMODULE_RN2XX3
+    default 1000
+    help
+        Configure the RX1 window delay (in milliseconds). This configuration
+        specifies the delay in seconds to open the RX1 window after the end of
+        uplink modulation on the end-node.
+
+choice
+    bool "RX2 window datarate index" if LORAMAC_ADVANCED_OPTIONS
+    depends on LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP || USEMODULE_RN2XX3
+    default LORAMAC_DEFAULT_RX2_DR_0 if LORAMAC_REGION_EU_868
+    default LORAMAC_DEFAULT_RX2_DR_2 if LORAMAC_REGION_IN_865
+    help
+        Configure the RX2 window datarate. This may be changed only if the
+        network server can be configured with the same datarate.
+
+config LORAMAC_DEFAULT_RX2_DR_0
+    bool "DR0"
+
+config LORAMAC_DEFAULT_RX2_DR_1
+    bool "DR1"
+
+config LORAMAC_DEFAULT_RX2_DR_2
+    bool "DR2"
+
+config LORAMAC_DEFAULT_RX2_DR_3
+    bool "DR3"
+
+config LORAMAC_DEFAULT_RX2_DR_4
+    bool "DR4"
+
+config LORAMAC_DEFAULT_RX2_DR_5
+    bool "DR5"
+
+config LORAMAC_DEFAULT_RX2_DR_6
+    bool "DR6"
+
+config LORAMAC_DEFAULT_RX2_DR_7
+    bool "DR7"
+
+if LORAMAC_REGION_US_915
+
+config LORAMAC_DEFAULT_RX2_DR_8
+    bool "DR8"
+
+config LORAMAC_DEFAULT_RX2_DR_9
+    bool "DR9"
+
+config LORAMAC_DEFAULT_RX2_DR_10
+    bool "DR10"
+
+config LORAMAC_DEFAULT_RX2_DR_11
+    bool "DR11"
+
+config LORAMAC_DEFAULT_RX2_DR_12
+    bool "DR12"
+
+config LORAMAC_DEFAULT_RX2_DR_13
+    bool "DR13"
+
+config LORAMAC_DEFAULT_RX2_DR_14
+    bool "DR14"
+
+config LORAMAC_DEFAULT_RX2_DR_15
+    bool "DR15"
+
+endif # LORAMAC_REGION_US_915
+
+endchoice
+
+config LORAMAC_DEFAULT_RX2_FREQ
+    int "RX2 Frequency (in Hz)" if LORAMAC_ADVANCED_OPTIONS
+    depends on USEMODULE_RN2XX3 || USEMODULE_GNRC_LORAWAN
```

Comment:
Should this depend on the selected frequency plan?
Can we apply ranges for the different plans? (`range <min> <max> if <region>`)

Reply:
Default RX2 frequency depends on the channel plan. The range on RX2 can technically be anything supported by the chipset/driver. The deciding factor, in reality, could be the gateways. The gateways usually come in four basic channel plans ( EU433, CN470, EU868 and US915), all other channel plans are derived. So we can have 4 different ranges.  @jia200x  any suggestions?

Reply:
Actually the default RX2 frequency depends of the network server. The frequency plan is not related, although there are "default" values for certain regions. As you said, this can be anything supported by the chipset driver. I would simply leave a 32 bit range here

Reply:
Makes sense. How can RX2 really depend on the network server if the gateway cannot support it? Don't you think gateway hence channel plan is the dependency here ?

Reply:
I suggest to not make this more complicate than it should be. To be precise, the RX2 range depend of the frequency plan. Within this range, the network server can decide the RX2 freq. So, it's not valid to assume this is dependent of the region.

 In practice this parameter is configured out of band with the required value of the network server anyway. The actual range or dependencies it might have won't add IMO more utilities than simply ensuring the app builds

Reply:
how about `range 137000000 1020000000` . That's the range of sx127x series chips

Reply:
ok, let's leave it like that and without the dependency to the frequency plan